lifecycle.go

```diff
@@ -62,62 +62,35 @@ func (l *lifecycle) Append(hook Hook) {
 	l.hooks = append(l.hooks, hook)
 }
 
-// start calls all OnStarts in order, halting on the first OnStart that fails
-// and marking that position so that stop can rollback.
-func (l *lifecycle) start() error {
+func (l *lifecycle) Start() error {
```

Comment:
Since the `Lifecycle` interface doesn't expose these methods, it's impossible for module authors to call them.